python/cuvs_bench/cuvs_bench/get_dataset/__main__.py

```diff
@@ -80,35 +89,141 @@ def download(name, normalize, ann_bench_data_path):
         raise
 
 
-def main():
-    call_path = os.getcwd()
-    if "RAPIDS_DATASET_ROOT_DIR" in os.environ:
-        default_dataset_path = os.getenv("RAPIDS_DATASET_ROOT_DIR")
-    else:
-        default_dataset_path = os.path.join(call_path, "datasets/")
-    parser = argparse.ArgumentParser(
-        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+def generate_ann_benchmark_like_data(
+    output_file="ann_benchmarks_like.hdf5",
+    n_train=1000,
+    n_test=100,
+    d=32,
+    centers=3,
+    k=100,
+    metric="euclidean",
+    dataset_path="test-data/",
+):
+    """
+    Generate a synthetic dataset in HDF5 format with a structure
+    similar to ann-benchmarks datasets. By default, ground truth
+    is computed for the top-100 nearest neighbors.
+    """
+
+    train_data, _ = make_blobs(
+        n_samples=n_train, n_features=d, centers=centers, random_state=42
     )
-    parser.add_argument(
-        "--dataset", help="dataset to download", default="glove-100-angular"
-    )
-    parser.add_argument(
-        "--dataset-path",
-        help="path to download dataset",
-        default=default_dataset_path,
+
+    test_data, _ = make_blobs(
+        n_samples=n_test, n_features=d, centers=centers, random_state=84
     )
-    parser.add_argument(
-        "--normalize",
-        help="normalize cosine distance to inner product",
-        action="store_true",
+
+    test_data = test_data.astype(np.float32)
+    train_data = train_data.astype(np.float32)
+
+    dist_matrix = cdist(test_data, train_data, metric=metric)
+
+    actual_k = min(k, n_train)
+    neighbors = np.argsort(dist_matrix, axis=1)[:, :actual_k].astype(np.int32)
+    distances = np.take_along_axis(dist_matrix, neighbors, axis=1).astype(
+        np.float32
     )
 
-    if len(sys.argv) == 1:
-        parser.print_help()
-        sys.exit(1)
-    args = parser.parse_args()
+    full_path = os.path.join(dataset_path, "test-data")
+    os.makedirs(full_path, exist_ok=True)
+    full_path = os.path.join(full_path, output_file)
+
+    with h5py.File(full_path, "w") as f:
+        # Datasets
+        f.create_dataset("train", data=train_data)
+        f.create_dataset("test", data=test_data)
+        f.create_dataset("neighbors", data=neighbors)
+        f.create_dataset("distances", data=distances)
 
-    download(args.dataset, args.normalize, args.dataset_path)
+        f.attrs["distance"] = metric
+
+    convert_hdf5_to_fbin(full_path, normalize=True)
```

Comment:
I wonder- should we add debug prints before `download` and `convert_hdf5_to_fbin` as well, just in case they fail?